src/opnsense/mvc/app/library/OPNsense/Backup/Nextcloud.php

```diff
@@ -137,7 +144,7 @@ public function backup()
             $backupdir = (string)$nextcloud->backupdir;
             $crypto_password = (string)$nextcloud->password_encryption;
             $hostname = $config->system->hostname . '.' .$config->system->domain;
-            $configname = 'config-' . $hostname . '-' .  date("Y-m-d_H:i:s") . '.xml';
```

Comment:
the easiest way is to omit ":" use here and pipe it through gettext() before passing to date(). at least that would be the sane way and language translations can still override it if needed. giving the user this option does not really add to usability especially since we assume it's a bug sitting somewhere between iso date and backup software writers on the other end of our "push".

Reply:
That’d be the easiest solution, indeed. There’s already a patch floating around doing just that. 

I’m still disputing that this is a problem of the software client (i.e. Nextcloud Client). I’d expect source.filename == destination.filename from any filetransfer software. It the dest FS can’t handle characters, that’s neither the fault of the source or the transfer utility. Who-/Whatever inserted those characters is.

Having an FAT / NTFS incompatible character hardcoded into a filename is just asking for trouble.

I want to empower users to enjoy their preferred date flavor, while keeping it compatible to their respective infrastructure.

Reply:
Let's keep this simple: we can't brace for all auxiliary impact. If aesthetically pleasing ISO dates are bad for any such reason we might as well use simple unix timestamps. I just don't think we'll ever be able to accommodate 8.3 in all of its restrictiveness so that is not a technical discussion to have in the first place.

Reply:
8.3 is auto generated these days, no reasons to worry about that.
Epoch is fine as well, just not as readable. 

Could you expand on what you mean by „auxiliary impact“?

Reply:
A seemingly simple choice for ISO date with 7bit ASCII characters leading to an incompatibility with a file system, how can anyone account for this... it even degrades readability on both ends and affects users who had no issues with this in the first place. But it is what it is. Would you mind updating and simplifying your PR?